kernel/kernel-api/src/main/java/io/delta/kernel/internal/columndefaults/ColumnDefaults.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.columndefaults;
+
+import io.delta.kernel.internal.actions.Metadata;
+import io.delta.kernel.internal.util.SchemaUtils;
+import io.delta.kernel.types.StructField;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utilities class for TableFeature "allowColumnDefaults". NOTE: As of Aug 2025, kernel only
+ * supports reading Delta tables with the table feature, or make metadata change to the table.
+ * Writing actual data to the table, or modify the default values is not allowed.
+ */
+public class ColumnDefaults {
+
+  public static String DEFAULT_VALUE_METADATA_KEY = "CURRENT_DEFAULT";
+
+  static class ColumnDefaultsCollector {
+    Map<String, String> collected = new HashMap<>();
+
+    void collect(List<String> paths, StructField field) {
+      if (field.getMetadata().contains(DEFAULT_VALUE_METADATA_KEY)) {
+        collected.put(
+            SchemaUtils.concatWithDot(paths),
+            field.getMetadata().getString(DEFAULT_VALUE_METADATA_KEY));
+      }
+    }
+  }
+  /** Block changes to column defaults stored in metadata */
+  public static void validateMetadataUpdate(Metadata oldMetadata, Metadata newMetadata) {
```

Comment:
do we allow the data type changes for default value columns as part of the schema evolution?